### PR TITLE
Fix AB for pre CEX7 cards

### PR DESCRIPTION
**Pull request by juergenchrist**
CI discovered that ECDH1 derivation does not work with AB keys on pre-CEX7 cards.  Fix and fail with CKR_MECHANISM_INVALID in that case.

### CODE REVIEW

File: testcases/crypto/abfunc.c [hunk 1]
```diff
@@ -565,24 +565,6 @@ void do_TestInvalidKeys(CK_SESSION_HANDLE session)
     CK_OBJECT_HANDLE pubkey = CK_INVALID_HANDLE, privkey = CK_INVALID_HANDLE;
     CK_RV rc;
 
-    // Expected to fail with CKR_TEMPLATE_INCOMPLETE
```
Comment (ifranzki) [inline testcases/crypto/abfunc.c:568]: So this removal is not related to the bug, right? 
Do I read it correctly that Dilithium keys are supported as AB keys?

Comment (juergenchrist) [reply testcases/crypto/abfunc.c:568]: Ups, this line also makes sense for the remaining test cases.  Putting it back in.  But yes, Dilithium keys are actually supported for AB keys (undocumented feature).  So if Dilithium can be used, this test case will actually fail.

**Review by ifranzki — APPROVED** [review]
lgtm

**Review by p-steuer — APPROVED** [review]
